var/spack/repos/builtin/packages/py-googleapis-common-protos/package.py

```diff
@@ -26,14 +28,20 @@ class PyGoogleapisCommonProtos(PythonPackage):
         description="Enable support for gRPC Remote Procedure Call framework.",
     )
 
-    with when("+grpc"):
-        depends_on("py-grpcio@1.44:1", when="@1.57:", type="run")
-        depends_on("py-grpcio@1", when="@1.56.2:", type="run")
-        depends_on("py-grpcio@1:", type="run")
-
     depends_on("py-setuptools", type="build")
-    depends_on(
-        "py-protobuf@3.19.5:3.19,3.20.2:4.21.0,4.21.6:4", when="@1.58:", type=("build", "run")
-    )
-    depends_on("py-protobuf@3.12.0:3", when="@1.55", type=("build", "run"))
-    depends_on("py-protobuf@3.6.0:3", when="@:1.6", type=("build", "run"))
+
+    with default_args(type=("build", "run")):
+        # https://github.com/googleapis/python-api-common-protos/blob/main/setup.py
+        # May be able to rais max version to :5 in next release
+        depends_on("py-protobuf@3.19.5:4", when="@1.58:")
+        depends_on("py-protobuf@3.15.0:4", when="@1.56:1.57")
+        depends_on("py-protobuf@3.12.0:4", when="@1.55")
+        depends_on("py-protobuf@3.6.0:", when="@1.6.0:")
+
+    # Explicitly incompatibile versions per setup.py
+    conflicts("py-protobuf@3.20:3.21.1")
```

Comment:
Based on `requires.txt`, this should be:

```suggestion
    conflicts("py-protobuf@3.20:3.20.1")
```